File: app-refresh/src/main/java/com/adammcneilly/cashcaretaker/data/Converters.java
package com.adammcneilly.cashcaretaker.data;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

@SuppressWarnings("WeakerAccess")
public class Converters {
    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        if (date == null) {
            return null;
        } else {
            return date.getTime();
        }
    }
}

File: database/src/main/java/com/androidessence/cashcaretaker/database/Converters.kt
package com.androidessence.cashcaretaker.database

import androidx.room.TypeConverter
import java.util.Date

/**
 * Database converters that will convert longs to dates and vise versa when inserting/pulling from the database.
 */
internal class Converters {
    @TypeConverter
    fun fromTimestamp(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? {
        return date?.time
    }
}